x-pack/docs/en/security/authentication/oidc-guide.asciidoc

```diff
@@ -0,0 +1,649 @@
+
+This will allow users that haven't already authenticated with OpenID Connect to
+navigate directly to the `/login` page in {kib} in order to use the login form.
+
+`xpack.security.auth.oidc.realm`::
+The name of the OpenID Connect realm in {es} that should handle authentication
+for this Kibana instance.
+
+`server.xsrf.whitelist`::
+{kib} has in-built protection against _Cross Site Request Forgery_ attacks which
+are designed to prevent the {kib} server from processing requests that
+originated from outside the {kib} application.
+In order to support OpenID Connect messages that originate from your
+OP or a third party (see <<third-party-login>>, we need to explicitly _whitelist_ the
+OpenID Connect authentication endpoint within {kib}, so that the {kib} server will
+not reject these external messages.
+
+
+=== OpenID Connect without {kib}
+
+The OpenID Connect realm is designed to allow users to authenticate to {kib} and as
+such, most of the parts of the guide above make the assumption that {kib} is used.
+This section describes how a custom web application could use the relevant OpenID
+Connect REST APIs in order to authenticate the users to {es}, with OpenID Connect.
+
+Single Sign On realms such as OpenID Connect and SAML make use of the Token Service in
+{es} and in principle exchange a SAML or OpenID Connect Authentication response for
+an {es} access token to be used as credentials for subsequent calls to {es} and a
+refresh token allowing to get new {es} access tokens for the user once the current one
+expires.
+
+NOTE: The {es} Token Service can be seen as a minimal oAuth2 authorization server
+and the access token and refresh token mentioned above are tokens that pertain
+_only_ to this authorization server, are generated and consumed _only_ by {es}
+and are in no way related to the tokens ( access token and ID Token ) that the
+OpenID Connect Provider issues.
+
+==== Register the RP with an OpenID Connect Provider
+
+The Relying Party ( {es} and the custom web app ) will need to be registered as
+client with the OpenID Connect Provider. Note that when registering the
+`Redirect URI`, it needs to be a URL in the custom web app.
+
+==== OpenID Connect Realm
+
+As before, an OpenID Connect realm needs to be created and configured accordingly
+in {es}. See <<oidc-guide-authentication>>
+
+==== Service Account user for accessing the APIs
+
+The realm is designed with the assumption that there needs to be a privileged entity
+acting as an authentication proxy. In this case, the custom web application is the
+authentication proxy handling the authentication of end users ( more correctly,
+"delegating" the authentication to the OpenID Connect Provider ). The OpenID Connect
+APIs require authentication and the necessary authorization level for the authenticated
+user. For this reason, a Service Account user needs to be created and assigned a role
+that gives them the `manage_oidc` cluster privilege. The use of the `manage_token`
```

Comment:
Does this cluster privilege need to be added to https://www.elastic.co/guide/en/elastic-stack-overview/master/security-privileges.html ?

Reply:
Thanks, I created https://github.com/elastic/stack-docs/pull/334